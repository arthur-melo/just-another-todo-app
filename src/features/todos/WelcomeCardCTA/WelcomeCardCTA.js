import CreateTodoItem from '../CreateTodoItem/CreateTodoItem';

const WelcomeCardCTA = () => (
  <div className="container">
    <div className="row justify-content-center">
      <div className="col-12 col-md-9 col-lg-7 col-xl-6">
        <div className="card text-center">
          <div className="card-body">
            <h5 className="card-title">Welcome!</h5>
            <p className="card-text">
              To get started, add some items to your list:
            </p>
            <div className="d-inline-block">
              <CreateTodoItem />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default WelcomeCardCTA;
