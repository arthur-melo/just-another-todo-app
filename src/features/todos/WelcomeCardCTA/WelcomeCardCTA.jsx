import CreateTodoItem from '../CreateTodoItem/CreateTodoItem';

const WelcomeCardCTA = () => (
  <div className="container">
    <div className="row justify-content-center">
      <div className="col-auto">
        <div className="card text-center border border-0">
          <div className="card-body">
            <h5 className="card-title">Welcome!</h5>
            <p className="card-text">
              To get started, add some items to your list:
            </p>
            <div className="container-fluid">
              <div className="row">
                <div className="mx-auto">
                  <CreateTodoItem />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default WelcomeCardCTA;
