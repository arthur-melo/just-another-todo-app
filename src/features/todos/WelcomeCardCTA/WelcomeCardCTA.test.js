import renderWithProviders from '../../../utils/testHelpers';
import WelcomeCardCTA from './WelcomeCardCTA';

describe('WelcomeCardCTA', () => {
  it('should render the WelcomeCardCTA component', () => {
    const view = renderWithProviders(<WelcomeCardCTA />);
    expect(view).toMatchSnapshot();
  });
});
