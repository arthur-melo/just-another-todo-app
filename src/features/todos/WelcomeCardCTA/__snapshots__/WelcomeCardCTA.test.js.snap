// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeCardCTA should render the WelcomeCardCTA component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row justify-content-center"
        >
          <div
            class="col-auto"
          >
            <div
              class="card text-center border border-0"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Welcome!
                </h5>
                <p
                  class="card-text"
                >
                  To get started, add some items to your list:
                </p>
                <div
                  class="container-fluid"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="mx-auto"
                    >
                      <form
                        autocomplete="on"
                        method="POST"
                      >
                        <div
                          class="d-flex"
                        >
                          <textarea
                            aria-label="Todo item description"
                            class="form-control create_todo_item__textarea"
                            id="new-todo-item"
                            name="new-todo-item"
                            placeholder="I want to do..."
                            type="text"
                          />
                          <div
                            class="mx-2"
                          >
                            <button
                              aria-label="Add todo item"
                              class="btn btn-primary"
                              disabled=""
                              title="Add todo item"
                              type="submit"
                            >
                              <svg
                                aria-hidden="true"
                                class="svg-inline--fa fa-plus fa-fw "
                                data-icon="plus"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                viewBox="0 0 448 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                                  fill="currentColor"
                                />
                              </svg>
                            </button>
                          </div>
                        </div>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-auto"
        >
          <div
            class="card text-center border border-0"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Welcome!
              </h5>
              <p
                class="card-text"
              >
                To get started, add some items to your list:
              </p>
              <div
                class="container-fluid"
              >
                <div
                  class="row"
                >
                  <div
                    class="mx-auto"
                  >
                    <form
                      autocomplete="on"
                      method="POST"
                    >
                      <div
                        class="d-flex"
                      >
                        <textarea
                          aria-label="Todo item description"
                          class="form-control create_todo_item__textarea"
                          id="new-todo-item"
                          name="new-todo-item"
                          placeholder="I want to do..."
                          type="text"
                        />
                        <div
                          class="mx-2"
                        >
                          <button
                            aria-label="Add todo item"
                            class="btn btn-primary"
                            disabled=""
                            title="Add todo item"
                            type="submit"
                          >
                            <svg
                              aria-hidden="true"
                              class="svg-inline--fa fa-plus fa-fw "
                              data-icon="plus"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              viewBox="0 0 448 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                                fill="currentColor"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
